src/pages/discovery/discovery.js

```diff
@@ -30,7 +30,8 @@ export default function Discovery() {
 	const context = useContext(UniversalContext);
 	const [selectedColumns, setSelectedColumns] = useState(initialiseSelectedColumnsState());
 	const [whereFilter, setWhereFilter] = useState(initialiseWhereFilterState());
-	const [isSortIconResetNeeded, setIsSortIconResetNeeded] = useState(false)
+	const [isSortIconResetNeeded, setIsSortIconResetNeeded] = useState(false);
+	const [dataLoaded, setDataLoaded] = useState(false);
```

Comment:
I would change the name, and the boolean value here so that it is clearer what is happening. I would call it something like showLoadingSpinner, and setShowLoadingSpinner. Default the state to true, and then set the state to false when loaded. Makes the code more readable.